app/controllers/sessions_controller.rb

```diff
@@ -11,16 +20,24 @@ def google_auth
         # Note: Refresh_token is only sent once during the first request
         refresh_token = access_token.credentials.refresh_token
         user.google_refresh_token = refresh_token if refresh_token.present?
+        # set appropriate permissions
+        user_is_admin = admins.include? access_token.info.email
+        user_is_staff = staff.include? access_token.info.email
         if existing_user
             session[:current_user_id] = user.id
             redirect_to root_path, flash: { :success => "Success! You've been logged-in!" }
         else
-            user.is_student = true  # new user is a student by default
+            user.is_student = (!user_is_admin and !user_is_staff)
+            user.is_admin = user_is_admin
+            user.is_staff = user_is_staff
             if user.save
                 if user.is_student
                     session[:current_user_id] = user.id
                     redirect_to login_confirm_path
-                else
+                elsif user.is_admin
+                    session[:current_user_id] = user.id
+                    redirect_to root_path, flash: { :success => "Success! You've been logged-in!" }
+                else # user must be staff
                     session[:current_user_id] = user.id
                     redirect_to root_path, flash: { :success => "Success! You've been logged-in!" }
```

Comment:
should we combine these two cases?

Reply:
I suppose we could for now, since staff signup isn't implemented yet, but I think the plan was to allow staff to also create a profile, which is why I allocated a separate path. Should I remove it until then?

Reply:
I think it can be left as is then. 👍